common/djangoapps/third_party_auth/tasks.py

```diff
@@ -36,7 +37,7 @@ def fetch_saml_metadata():
         num_failed: Number of providers that could not be updated
         num_total: Total number of providers whose metadata was fetched
     """
-    if not SAMLConfiguration.is_enabled():
+    if not SAMLConfiguration.is_enabled(get_current_site_default()):
```

Comment:
This task runs during times when there is no request context (saml management command, and I also think it runs on startup), so the default site fallback is necessary here.